pointcloud/basic_octree_utils.go

```diff
@@ -165,11 +161,38 @@ func (octree *basicOctree) helperIterate(lowerBound, upperBound, idx int, fn fun
 			idx += numPoints
 		}
 
-	case LeafNodeFilled:
+	case leafNodeFilled:
 		ok = fn(octree.node.point.P, octree.node.point.D)
 
-	case LeafNodeEmpty:
+	case leafNodeEmpty:
 	}
 
 	return ok
 }
+
+// ConvertPointCloudToBasicOctree converts a pointcloud into a octree pointcloud representation.
+func ConvertPointCloudToBasicOctree(cloud PointCloud) (PointCloud, error) {
```

Comment:
I would expect this to return a BasicOctree based on the function name. Why not return a BasicOctree here?

Reply:
Agreed

Reply:
I initially went with a two stage approach so support future octree implementations. Therefore the function `ReadPCDToBasicOctree` wraps this as could future implementations. Possible options include:

1. Rename function to  `ConvertPointCloudToOctree`
2. Change return type to *BasicOctree (would we want to keep `ReadPCDToBasicOctree` where it is in pointcloud_file)?
3. Create an additional wrapper `ReadPCDToBasicOctree` that calls `ReadPCDToOctree`

Reply:
I think we can just change the return type of this. I'm not sure how likely we are to have more implementations of Octree.